File: Frontend/institution_web_app/lib/models/Korisnik.dart
import 'Grad.dart';

class Korisnik{
  int id;
  String email;
  String ime;
  String prezime;
  String biografija;
  String username;
  String password;
  String urlSlike;
  String uloga;
  int poeni;
  List<Grad> gradovi = List<Grad>();
  Korisnik({this.email, this.ime, this.prezime, this.biografija,this.username,this.password,this.urlSlike,this.uloga,this.poeni,this.gradovi});

  factory Korisnik.fromJson(Map<String, dynamic> json){
    return Korisnik(
        email: json['email'],
      ime: json['ime'],
      prezime: json['prezime'],
      biografija: json['biografija'],
	    username: json['username'],
      password: json['password'],
	    urlSlike: json['urlSlike'],
      uloga: json['uloga'],
      poeni: json['poeni']

    );
  }

  Map<String, dynamic> toMap(){
    var map = Map<String, dynamic>();
    map["email"] = email;
    map["ime"] = ime;
    map["prezime"] = prezime;
    map["biografija"] = biografija;
  	map["username"] = username;
    map["password"] = password;
    map["urlSlike"] = urlSlike;
    map["uloga"] = uloga;
    map["poeni"] = poeni;

    if(id != null){
      map["id"] = id;
    }
    return map;
  }

  Korisnik.fromObject(dynamic o){
    this.id = o["id"];
    this.email = o["email"];
    this.ime = o["ime"];
    this.prezime = o["prezime"];
    this.biografija = o["biografija"];
  	this.username = o["username"];
    this.password = o["password"];
  	this.urlSlike = o["urlSlike"];
    this.uloga = o["uloga"];
    this.poeni = o["poeni"];
    if(o["gradovi"]!=null){
      var rest = o["gradovi"] as List;
      this.gradovi = rest.map<Grad>((e) => Grad.fromObject(e)).toList();
    }
  }

}

File: Backend/MyCity_backend/Models/Korisnik.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace backendPoSlojevima.Models
{
    public class Korisnik
    {
        public long id { get; set; }
        public String email { get; set; }
        public String username {get; set;}
        public String password {get; set;}
        public String ime { get; set; }
        public String prezime { get; set; }
        public String biografija { get; set; }
        public String Token { get; set; }
        public String uloga { get; set; }
        public String urlSlike { get; set; }
        public long poeni { get; set; }
        public int ocenaAplikacije { get; set; }
        //public long validacija { get; set; }
       // public virtual List<Objave> objave { get; set; }
    }
}
